### PR TITLE
fix makefile install and deploy targets

### PR DESCRIPTION
**Pull request by mmartinv**
- Fix install and deploy targets
- Improve Makefile readability
- Rename deprecated field name

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -96,12 +96,12 @@ run: generate fmt vet
...
 # Install CRDs into a cluster
 install: manifests
-	kubectl apply -f config/crds
+	kubectl apply -f operator/config/crd/bases
 
 # Deploy controller in the configured Kubernetes cluster in ~/.kube/config
 deploy: manifests
-	kubectl apply -f config/crds
-	kustomize build config/default | kubectl apply -f -
+	kubectl apply -f operator/config/crd/bases
```
Comment (ahadas) [inline Makefile:103]: I don't get something here - isn't it enough to have the command below?

**Review by ahadas — COMMENTED** [review]
I fixed deply to deploy in the PR's description - please change the commit message of the third commit as well

File: Makefile (hunk 1 above)
Comment (mmartinv) [reply Makefile:103]: Yeah, according to `operator/config/default/kustomization.yml`  which by the way has the `bases` deprecated field. 
Sorry I didn't look deeper I just adjusted the actual directories, I think both targets could be simplified with the following, right?:
```
install: manifests
	kustomize build operator/config/crd  | kubectl apply -f -

# Deploy controller in the configured Kubernetes cluster in ~/.kube/config
deploy: manifests
	kustomize build operator/config/default | kubectl apply -f -
```

Comment (ahadas) [reply Makefile]: yeah

File: Makefile [hunk 2]
```diff
@@ -95,13 +100,12 @@ run: generate fmt vet
...
-	kubectl apply -f config/crds
+install: manifests kubectl
+	$(KUBECTL) apply -k operator/config/crds
 
 # Deploy controller in the configured Kubernetes cluster in ~/.kube/config
-deploy: manifests
-	kubectl apply -f config/crds
-	kustomize build config/default | kubectl apply -f -
+deploy: manifests kubectl
+	$(KUBECTL) apply -k operator/config/default
```
Comment (ahadas) [inline Makefile:108]: should be `kustomize build operator/config/default | $(KUBECTL) apply -f `, no?

**Review by ahadas — APPROVED** [review]
the `deploy` target was broken already so let's get it in but @mmartinv please take a look at the comment inside

File: Makefile (hunk 2 above)
Comment (mmartinv) [reply Makefile:108]: That's equivalent to `$(KUBECTL) apply -k operator/config/default`

From `kubectl apply -h`:
```
    -k, --kustomize='':
	Process a kustomization directory. This flag can't be used together with -f or -R.
```

**Pull request merged**